src/main/scala/mesosphere/marathon/core/launchqueue/impl/LaunchQueueActor.scala

```diff
@@ -0,0 +1,186 @@
+  var launcherRefs = Map.empty[ActorRef, PathId]
+
+  /** Serial ID to ensure unique names for children actors. */
+  var childSerial = 0
+
+  // See [[receiveHandlePurging]]
+  /** A message with a sender for later processing. */
+  case class DeferredMessage(sender: ActorRef, message: Any)
+  /** PathIds for which the actors have been currently suspended because we wait for their termination. */
+  var suspendedLauncherPathIds = Set.empty[PathId]
+  /** ActorRefs of the actors have been currently suspended because we wait for their termination. */
+  var suspendedLaunchersMessages = Map.empty[ActorRef, Vector[DeferredMessage]].withDefault(_ => Vector.empty)
+
+  /** The timeout for asking any children of this actor. */
+  implicit val askTimeout: Timeout = 3.seconds
+
+  override def receive: Receive = LoggingReceive {
+    Seq(
+      receiveHandlePurging,
+      receiveMessagesToSuspendedActor,
+      receiveHandleNormalCommands
+    ).reduce(_.orElse[Any, Unit](_))
+  }
+
+  /**
+    * Handles purging of an actor.
+    *
+    * Terminating an actor with a PoisonPill is not instantaneous. It will still process
+    * all prior messages. While waiting for the termination of the actor, we might
+    * receive further messages to the actor which would potentially lead to recreating it.
+    *
+    * Thus, we mark the actor as suspended and save all messages which would normally
+    * have been sent to this actor. When we receive confirmation of the actor's death (Terminated),
+    * we will replay these messages to ourselves with the correct sender.
+    */
+  private[this] def receiveHandlePurging: Receive = {
+    case Purge(appId) =>
+      launchers.get(appId) match {
+        case Some(actorRef) =>
+          val deferredMessages: Vector[DeferredMessage] =
+            suspendedLaunchersMessages(actorRef) :+ DeferredMessage(sender(), ConfirmPurge)
+          suspendedLaunchersMessages += actorRef -> deferredMessages
+          suspendedLauncherPathIds += appId
+          actorRef ! PoisonPill
+        case None => sender() ! (())
+      }
+
+    case ConfirmPurge => sender() ! (())
+
+    case Terminated(actorRef) =>
+      launcherRefs.get(actorRef) match {
+        case Some(pathId) =>
+          launcherRefs -= actorRef
+          launchers -= pathId
+
+          suspendedLaunchersMessages.get(actorRef) match {
+            case None =>
+              log.warning("Got unexpected terminated for app {}: {}", pathId, actorRef)
+            case Some(deferredMessages) =>
+              deferredMessages.foreach(msg => self.tell(msg.message, msg.sender))
+
+              suspendedLauncherPathIds -= pathId
+              suspendedLaunchersMessages -= actorRef
+          }
+        case None =>
+          log.warning("Don't know anything about terminated actor: {}", actorRef)
+      }
+  }
+
+  private[this] def receiveMessagesToSuspendedActor: Receive = {
+    case msg @ Count(appId) if suspendedLauncherPathIds(appId) =>
```

Comment:
The launchQueueDelegate is sending these messages and it should be ok. We just wait for the actor to die and the process all messages. It is possible that we replay messages that add tasks to that app again, that's why we cannot answer that immediately.

We want to give the appearance of processing everything serially which relates to one app.